test/blackbox-tests/test-cases/pkg/compute-checksums-when-missing.t

```diff
@@ -1,87 +1,131 @@
 Test that dune will add checksums to lockfiles when the package has a source
-archive but no checksum.
+archive but no checksum. This test creates small http servers to serve packages
+to test checksum generation, since we only generate checksums for packages
+downloaded from non-local sources.
   $ . ./helpers.sh
   $ mkrepo
 
-  $ strip_pwd() {
-  >   sed -e "s#$PWD#<pwd>#"
+Run the http servers on a random port to reduce the chance of colliding with
+the port used by some other random server on this machine.
+  $ PORT=$(shuf -i 10000-65535 -n 1)
+
+  $ strip_transient() {
+  >   sed -e "s#$PWD#<pwd>#" | \
+  >   # strip the entire address as the port number may also appear in the hash
+  >   sed -e "s#http://.*$PORT#<addr>#"
   > }
 
+Make a webserver that serves a single file once before terminating:
+  $ cat >oneshot_webserver.ml <<EOF
```

Comment:
How about turning this into a proper binary and using it as follows:

```
$ serve_once --file FILE --port-file FILE
```

This will serve `--file` and then terminate. The program will select a port randomly and write the selection `--port-file`.

Reply:
Oh and this way we won't need to depend on ocamlfind for this test

Reply:
I moved the server to test/blackbox-tests/utils